src/new/wallet.js

```diff
@@ -342,21 +342,18 @@ class HathorWallet extends EventEmitter {
    * @return {Promise} Promise that resolves when transaction is sent
    **/
   sendPreparedTransaction(data) {
+    storageProxy.setStorage(this.storage);
     return transaction.sendPreparedTransaction(data);
   }
 
   /**
    * Connect to the server and start emitting events.
    **/
   start() {
-    const store = new MemoryStore();
-    storage.setStore(store);
-    storage.setItem('wallet:server', this.server);
-
-    ws.on('is_online', this.onConnectionChange);
-    ws.on('reload_data', this.reloadData);
-    ws.on('addresses_loaded', this.onAddressesLoaded);
-    ws.on('wallet', this.handleWebsocketMsg);
+    storageProxy.setStorage(this.storage);
+    this.conn.start();
```

Comment:
> In light of the next comment I just made, I think it's better to always call `conn.start()` so we can handle the result. If it's already started, it should resolve the promise immediately.

I agree.

Reply:
When I was doing this PR my thoughts was that we connection was not going to be shared in this first version. We need to think about bigger problems (not just start/stop) if we decide to allow shared connections in this PR.

For instance, one of the problems not solved in this PR is that two HathorWallet instances would receive the event for one new tx, which instance should update the storage?

That's why I've decided not to handle this case now.

Reply:
All right, I agree we postpone connection sharing for now.

Reply:
I agree we can postpone because we have to test all these cases. But, specifically about your point, I think all wallets will receive all transactions and they will filter them. I'm not sure if this criteria is enough, but a starting filter criteria would be to have at least one input or output that contains an address of that wallet.

Reply:
I agree about the wallets, however nowadays we already update some storage data from `WebSocketHandler`, which would receive events from more than one wallet. We still need to validate that all event handled by this class can share the storage.